docs/api/dns.md

```diff
@@ -8,34 +8,106 @@ console.log(addrs);
 // => [{ address: "172.67.161.226", family: 4, ttl: 0 }, ...]
 ```
 
-<!--
-## `Bun.dns` - lookup a domain
-`Bun.dns` includes utilities to make DNS requests, similar to `node:dns`. As of Bun v0.5.0, the only implemented function is `dns.lookup`, though more will be implemented soon.
-You can lookup the IP addresses of a hostname by using `dns.lookup`.
+## DNS caching in Bun
+
+In Bun v1.1.9, we added support for DNS caching. This cache makes repeated connections to the same hosts faster. 
+
+At the time of writing, we cache up to 255 entries for a maximum of 30 seconds (each). If any connections to a host fail, we remove the entry from the cache. When multiple connections are made to the same host simultaneously, DNS lookups are deduplicated to avoid making multiple requests for the same host.
+
+This cache is automatically used by;
+
+- `bun install`
+- `fetch()`
+- `node:http` (client)
+- `Bun.connect`
+- `node:net`
+- `node:tls`
+
+### When should I prefetch a DNS entry?
+
+Web browsers expose [`<link rel="dns-prefetch">`](https://developer.mozilla.org/en-US/docs/Web/Performance/dns-prefetch) to allow developers to prefetch DNS entries. This is useful when you know you'll need to connect to a host in the near future and want to avoid the initial DNS lookup.
+
+In Bun, you can use the `dns.prefetch` API to achieve the same effect. 
+
 ```ts
-import { dns } from "bun";
-const [{ address }] = await dns.lookup("example.com");
-console.log(address); // "93.184.216.34"
+import {dns} from "bun";
+
+dns.prefetch("my.database-host.com", 5432);
 ```
-If you need to limit IP addresses to either IPv4 or IPv6, you can specify the `family` as an option.
+
+An example where you might want to use this is a database driver. When your application first starts up, you can prefetch the DNS entry for the database host so that by the time it finishes loading everything, the DNS query to resolve the database host may already be completed.
+
+### `dns.prefetch`
+
+{% callout %}
+**🚧** — This API is experimental and may change in the future.
+{% /callout %}
+
+To prefetch a DNS entry, you can use the `dns.prefetch` API. This API is useful when you know you'll need to connect to a host in the soon and want to avoid the initial DNS lookup.
```

Comment:
```suggestion
To prefetch a DNS entry, you can use the `dns.prefetch` API. This API is useful when you know you'll need to connect to a host soon and want to avoid the initial DNS lookup.
```